core/arch/arm/kernel/elf_load.c

```diff
@@ -161,13 +164,22 @@ static TEE_Result copy_to(struct elf_load_state *state,
 	if (!len)
 		return TEE_SUCCESS;
 
+	/* Check for integer overflow */
+	if ((len > SIZET_MAX - dst_offs) || ((size_t)dst > SIZET_MAX - dst_offs))
+		return TEE_ERROR_SECURITY;
+
 	if (len > dst_size || (len + dst_offs) > dst_size)
```

Comment:
Wouldn't changing `len > dst_size` to `(len + dst_offs) < dst_offs` here make all the other added checks in this function redundant?